### ISSUE TITLE
ReferenceError: HTMLElement is not defined

### ISSUE THREAD

**Issue by iAmVaeVictus** [olifolkerd/tabulator#1989]
@olifolkerd - Was there a solution to this closed issue (https://github.com/olifolkerd/tabulator/issues/1877)? I am experiencing the same thing with Tabulator (v4.2.5), node.js (6.9.0), express (4.16.0), handlebars (4.1.1), jQuery (1.12.4).

index.js
-----
```
var Tabulator = require('tabulator-tables');
var sampleData= [
        {id:1, name:"Oli Bob"},
        {id:2, name:"Mary May"},
];

var table = new Tabulator('#example-table", {
  bla bla bla copied from the sample code

table.setData(sampleData);
```
package.json contains:
```
{"name": "01-phati-express-dev",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "start": "node ./bin/www"
  },
  "dependencies": {
    "chopped-and-viewed": "^1.0.0",
    "cookie-parser": "~1.4.3",
    "debug": "~2.6.9",
    "express": "~4.16.0",
    "express-handlebars": "^3.0.2",
    "fast-csv": "^2.4.1",
    "fs": "0.0.1-security",
    "http-errors": "~1.6.2",
    "morgan": "~1.9.0",
    "os": "^0.1.1",
    "path": "^0.12.7",
    "string": "^3.3.3",
    "tabulator-tables": "^4.2.5",
    "url": "^0.11.0",
    "util": "^0.11.1"
  }}
```

index.hbs 
`<div id="example-table"></div>`

Error is at the "new" statement.

Stack:
```
./tabulator-tables/dist/js/tabulator.js:7257
		if (element instanceof HTMLElement) {
		                       ^
ReferenceError: HTMLElement is not defined
    at Tabulator.initializeElement (../tabulator-tables/dist/js/tabulator.js:7257:26)
    at new Tabulator (../tabulator-tables/dist/js/tabulator.js:6782:8)
    at Object.<anonymous> (./routes/index.js:28:17)
    at Module._compile (module.js:660:30)
    at Object.Module._extensions..js (module.js:671:10)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Module.require (module.js:604:17)
    at require (internal/module.js:11:18)

```

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] - Dev on MacOS Mojave 10.14.4
 - Browser [e.g. chrome, safari] - Brave, Chrome

**Comment by takuy** [comment]
This sounds like like an issue with trying to using Tabulator on the server...? This is a client library.

**Comment by iAmVaeVictus** [comment]
Yes, I see what you are saying - and yes, this is my first web app, so I'm stumbling around in the dark, but, this code is in the index.js routes file that sends the info to the index.hbs (handlebars). Is the .js file server side in that case? 

Oh, and thanks for the reply!

**Comment by takuy** [comment]
Yeah, generally in node projects, whatever handles the routes, or running the server itself, is all happening server-side. So, in this case, index.js is the backend or the server.

Traditionally, JavaScript has been a client-side language and is still used as such to power most of the great things you see on the internet. Client-side javascript is usually included either on the same page as your rendered HTML or served in file which is then included in your HTML using `<script>` tags. The latter method is generally preferred as it allows for caching and is served in a separate file. 

Your handlebars template (index.hbs) are created on the server, rendered/transformed into HTML and served to the client for their browser. Client-side javascript can interact with this HTML once it's someone's browser. 
Therefore, with your template, you should learn how to include client-side javascript files.  There are several examples in this StackOverflow article on how to do that: https://stackoverflow.com/questions/17722407/how-to-include-javascript-on-client-side-of-node-js but none are Handlebars specific. 

This tutorial: https://hackersandslackers.com/handlebars-templating-in-expressjs/ seems to be a pretty good starting point for Handlebars/Express/Node stuff in general and you should be able to work the two examples together.

Basically, you'll want to do the following:
create a javascript file for the client (node tutorials will generally include this as part of the "views")
figure out how to serve this in your routes (see stackoverflow link)
include this in your template file (should be between the `<head>` tags - see the second tutorial)
use tabulator in your client-side javascript file

I'm not a big Node guy but this should doable and should be a great learning experience to see the difference between client and server-side javascript.

**Comment by iAmVaeVictus** [comment]
Thanks @takuy I appreciate the in depth explanation. I wrote an app in Node that ran synchronously, and was command line based. Converting it now to web, asynchronous, lots to learn here. But man, coding now in these high level languages vs C/C++ etc from 20 years ago - so much faster!
